---
title: "COO-I-4. Binning"
start: false
teaching: 0
exercises: 20
questions:
- How to cluster contigs into bins in a metagenomic assembly?
objectives:
- Gain experience with binning software output
- Interpret completeness and contamination scores associated to metagenomic bins 
---

## Binning

Next, we need to classify these contigs into bins. To do this, multiple binning software are available. 
MetaWRAP includes a complex binning pipeline that employs multiple binning software and evaluates the results jointly. 
Unfortunately, again, most tools require long runtimes. To avoid this inconvenience, Metawrap was already ran,
employing the binning software Concoct, maxbin2 and metabat2. Again, fetch the results from the appropriate folder

~~~
$ cd ~/GenomeBioinformatics/Block1/COO-I/Results/
$ mkdir 05_binning
$ cd 05_binning
$ ln -s ~/data_bb3bcg20/Block1/COOI/Intermediate_files/Binning/metawrap/ .
~~~
{: .bash}

> ## Exercise: How many bins were generated by Concoct, maxbin2 and metabat2
> 
>> ## Solution
>> `ls -l *_bins/`
>> 
>> `for f in *_bins; do echo $f; ls $f/*.fa | wc -l; done`
>> 
> {: .solution}
{: .challenge}

## Assessing bin quality

MetaWRAP assesses the quality of each bin using CheckM. It accomplishes this by comparing the contigs within 
each bin to a comprehensive database of gene markers from known reference genomes, thus evaluating the completeness, 
contamination, and strain heterogeneity scores of each bin. 
Completeness indicates the proportion of gene markers that is present in the bin. Since these gene markers are meant
to be essential genes, this score is often used as a proxy to how much of the genome has been recovered. 
Contamination measures the presence of sequences from other organisms within the bin, 
highlighting potential inaccuracies in the binning process. Strain heterogeneity assesses the genetic variability within 
a bin, which is crucial for distinguishing closely related strains. 

The files with suffix “.stats” include CheckM scores, plus a predicted taxonomical identification of each bin based on 
similarity to reference sequences. Compare the results of the three binning software.

> ## Exercise: What is the expected quality of the obtained bins?
> - Which software provided more high-quality bins?
>
> - Do you expect any of these MAGs may be represented by a complete, single-contig assembly?
>
> Note: for an easier visualisation of tab-delimited .tsv files in the command line, use the `less` wrapper `coltab.sh`:
> 
> `$ ~/data_bb3bcg20/bin/scripts/coltab.sh $file`
> 
>> ## Solution
>> `for f in *stats; do echo $f; cat $f; echo; done | ~/data_bb3bcg20/bin/scripts/coltab.sh`
>>
>> All three tools recover 2 bins with over 99% completeness scores and very low contamiination scores. These correspond
>> to bins classified by CheckM within the genus Clostridium (bacterium) and the phylum Euryarchaeota (archaea). In all
>> three cases, the Euryarchaeota genome is formed by one contig that stretches the entirety or almost the entirety of the
>> bin (N50 is equal or almost equal to the total size). Besides these two bins, all three binning tools recover additional
>> bins, but with relatively poor completeness scores.
>> 
> {: .solution}
{: .challenge}
