---
title: "Block 3 - COOII: Large-scale chromosomal variation in *Saccharomyces cerevisiae* (part 1)"
start: false
teaching: 0
exercises: 90

questions:
- How can you we identify large-scale genomic differences by directly comparing genome assemblies?

objectives:
- Perform whole-genome alignments, both between different genome assemblies as well as between different strains of the same species.
- Interpret whole-genome alignments to identify miss-assemblies as well as structural varients;

keypoints:
- You can work with and compare genome assemblies.  
- You can detect large-scale genomic differences by interpreting whole-genome comparisions.
---

# Large-scale chromosomal variation in Saccharomyces cerevisiae
Genome sequences of individuals of the same species can differ significantly, e.g. two humans are expected to differ at 0.5% of the nucleotides. Next to these single nucleotide differences, genomes can differ by larger structural variations such as translocations, inversions, duplications, or deletions. Mobile genetic element (transposon) insertions and deletions can further establish genetic diversity. Collectively, these large-scale variations often affect hundreds of nucleotides can significantly impact the phenotype.

Bakersâ€™ yeast has been traditionally used as a model to study how genotypic variations is mechanistically established. The availability of sequencing data of hundreds of different yeast isolates enables us to study the emergence and consequences of structural variation in detail. We here will work with assembled genome sequences (you have learnt how genomes are assembled in Block 1).

## 1. Comparision between two *S. cerevisiae* reference genome assemblies 
In the first episode of this excercise, we are going to perform whole-genome alignments to identify differences between genome assemblies of different yeast strains. Whole-genome alignments are often used as starting points to identify conserved regions as well as large-scale differences between genomes such as deletions or insertions, as well as chromosomal rearrangements such as translocations and inversions.

We will be generating whole-genome alignments using the [MUMMER](https://mummer.sourceforge.net) package. Depending on the expected similarity between species or strains, MUMMER allows to perform nucleotide-based (`nucmer`) or translated nucleotide-based alignemts (`promer`). These alignments can be analyzed graphically identifying qualitiative differences between the two genomes. For example, one can easily spot large-scale rearrangements such as translocations and insertions by inspecting a dotplot of depicting the alignments. Furthermore, the alignments generated by MUMMER can also be analyzed quantitatively to systematically search the genomic coordinates of the alignments to identify genetic differences ranging from single-nucleotide polymorphisms to large-scale variations (that could be visible in an graphical representation as well).

As you know from Block 1, sequence assembly is the process of reconstructing of long contiguous genomic sequences (contigs or scaffolds, ideally chromosomes) from short sequencing reads. Genome assemblies are not guaranteed to be correct and at best represent an informed hypothesis about the real genome sequence of an individual. *S. cerevisiae* strain S288C had been selected to be the initial yeast strain used to determine the genome sequence of this important model species (see the [Yeast Genome paper](https://www.science.org/doi/10.1126/science.274.5287.546). It is a laboratory strain commonly used in yeast research. Since it's initial publication in 1996, the S288C reference genome assembly has been continiously refined to improve its quality and contiguity.

We will compare the S288C reference genome assembly with a second assembly of the same strain that was recently established. The newer genome assembly was based on 3rd generation long-read sequencing technology (see paper by [Yue and colleagues](https://www.nature.com/articles/ng.3847), and thus we anticipate that it is of higher quality.

To test this hypothesis, we will align the two independent genome assemblies using nucmer to identify and study the differences between this two genome assemblies.

Use commands including ls and pwd to localize the two S288C genome assemblies (S288C.genome.fa and Yue2017_S288C.genome.fa) in the data storage folder. Then create a symbolic link to your own folder with the ln -s command.


> ## Exercise
>
> Take a look the parameters nucmer and take a consult the manual to find the explanation of the prarameters `nucmer -h`.
> - What is the difference between --mum, --mumreference, and --maxmatch options?
> - What could be the impact of the different options when interpreting the genome alignments?
>
>> ## Solution
>> `--mum` will only find maximum unique matches that are unique in the reference and in the query, this is the original definition of mummer from 1999. `--mumreference` will return matches unqiue in the reference sequence only (that is the default) and `--maxmatch` will return matches irrespective of their uniqueness in both reference and query. Thus, `--mum` and `--mumreference` will only return matches if they are unique, i.e., if duplications or other repetetive elements (such as transposons) occur, the alignments between these regions are not reported.
>>
> {: .solution}
{: .challenge}

We can now execute a `nucmerz command to calculate a whole-genome alignment using the S288C original reference genome (S288C.genome.fa) as the reference and the S288C long-read assembly (Yue2017_S288C.genome.fa) as a query. We retained all possible matches (`--maxmatch`) between the query and the reference and provided a prefix to give your results a clear name `-p`.

~~~
$ nucmer --maxmatch -p S288CvS288Cpb S288C.genome.fa Yue2017_S288C.genome.fa
~~~
{: .bash}

In a second step, we can now visually inspect the whole-genome alignment using `mummerplot`. `mummerplot` generates a dotplot to visualize the similarity between genomic regions; genomic regions that are similar when comparing the two genomes are shown by a dot. This dotplot allows to rapidly visualize large-scale differences between the two genomes (e.g. deletions or insertions) as well as chromosomal rearrangements (e.g. translocations or inversions).

First, note the parameters of `mummerplot`, and check what input file `mummerplot` expects and how you can modify the output visualization, e.g., via `--color`. 

~~~
$ mummerplot --color S288CvS288Cpb.delta --png
~~~
{: .bash}

> ## Exercise
>
> Transfer a copy of the file to your own machine. What do you oberserve (in general) when you look at the figure. Is this according to your expectations (remember what genome sequences did you align)?
>
>> ## Solution
>>
>> The alignment displays overall very high co-linearity (order and content is conserved) and no major differences can be observed. This is expected as two assemblies of the same yeast strains are compared. There are some obvious differences at the chromosomal ends, likely telomeres, which are typically complex and very challanging to assemble (Do you know why?)
>> 
> {: .solution}
{: .challenge}

